app/models/subscription.rb

```diff
@@ -51,11 +53,30 @@ def new_comment_by_someone_else?
   end
 
   def last_author
-    (discussion.comments.last || discussion).author
+    (last_comment || discussion).author
   end
 
-  def notify
-    NotificationsMailer.new_comment(id).deliver_later
+  def last_comment
+    discussion.comments.last
+  end
+
+  def notify_new_discussion
+    return unless new?
+    user.notifications.new_discussion(discussion)
```

Comment:
Yes, but this way you're leveraging your knowledge of the user's internals. It's a trade off. 

I'd prefer not to break LOD, but it's up to you.

> On 22 Jan 2015, at 00:00, Miguel Palhas <notifications@github.com> wrote:
> 
> In app/models/subscription.rb:
> 
> >    end
> >  
> > -  def notify
> > -    NotificationsMailer.new_comment(id).deliver_later
> > +  def last_comment
> > +    discussion.comments.last
> > +  end
> > +
> > +  def notify_new_discussion
> > +    return unless new?
> > +    user.notifications.new_discussion(discussion)
> that was the idea, and I believe the first version was similar to that, but it would require me to pass the user_id as a parameter as well. This way that's implicit
> 
> —
> Reply to this email directly or view it on GitHub.
>